Expensify/App
# [No QA] Skip IOU participants step if launched from within chat

### Details
- Skip the `IOUParticipants` if routed from a chat
- Dependent on https://github.com/Expensify/Expensify.cash/pull/2120 being merged

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2246

### Tests
- Add the following imports in `src/pages/home/report/ReportActionCompose.js`:
```
import Navigation from '../../../libs/Navigation/Navigation';
import ROUTES from '../../../ROUTES';
```

- Add the following to the menuItems in `src/pages/home/report/ReportActionCompose.js`:
```
{
    icon: Paperclip,
    text: 'Request Money',
    onSelected: () => {
        Navigation.navigate(
            ROUTES.getIouRequestRoute(this.props.reportID),
        );
    },
},
{
    icon: Paperclip,
    text: 'Split Bill',
    onSelected: () => {
        Navigation.navigate(
            ROUTES.getIouSplitRoute(this.props.reportID),
        );
    },
},
```
- Replace IOU_REQUEST and  IOU_BILL with the following in `src/ROUTES.js`:
```
    IOU_REQUEST: 'iou/request/:reportID',
    getIouRequestRoute: reportID => `iou/request/${reportID}`,
    IOU_BILL: 'iou/split/:reportID',
    getIouSplitRoute: reportID => `iou/split/${reportID}`,
```
    
- Go into any chat, click the "+" icon and click one of the above menu items
- Go through the IOU steps and confirm IOUParticipants step was skipped
- Then remove `this.props.reportID` from the navigation request above and go through the flow again, this time confirming that the IOUParticipants step does appear.

### QA Steps
Same as above.

### Tested On


### Screenshots

#### Web

https://user-images.githubusercontent.com/12075600/115362237-2e4dde00-a176-11eb-8356-6533814fb30a.mov

#### Mobile Web

https://user-images.githubusercontent.com/12075600/115362288-39087300-a176-11eb-8509-26aef387f524.mov

#### Desktop

https://user-images.githubusercontent.com/12075600/115362330-458ccb80-a176-11eb-9279-c42d8c6eca88.mov

#### iOS

https://user-images.githubusercontent.com/12075600/115362401-58070500-a176-11eb-8e95-297a3657a094.mov

#### Android

https://user-images.githubusercontent.com/12075600/115362454-66552100-a176-11eb-8558-ab2d0c279e68.mov
